docs/guides/docs/intro-to-sway/rust-sdk.mdx

```diff
@@ -0,0 +1,323 @@
+---
+title: Rust Testing
+category: Intro to Sway
+parent:
+  label: Guides
+  link: /guides
+---
+
+## Testing the contract
+
+You can find the full contract code, along with example tests using the Rust SDK, in the provided repository.
+
+### Generating a Test Template in Rust
+
+To create your own test template using Rust, follow these steps with `cargo-generate`:
+
+1. Install `cargo-generate`:
+```bash
+cargo install cargo-generate
+```
+
+2. Generate the template:
+```bash
+cargo generate --init fuellabs/sway templates/sway-test-rs --name contract
+```
+
+### Running the Tests
+
+For running tests located in `harness.rs`, use:
+```bash
+cargo test
+```
+
+If you wish to print outputs to the console during tests, execute:
+```bash
+cargo test -- --nocapture
+```
+
+## Imports
+
+At the beginning of the harness.rs file, import the Fuel Rust SDK. The majority of the functionalities you'll need are housed within the prelude.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_import"
+  commentType="//"
+  lang="rust"
+/>
+
+Always compile your contracts after making any changes. This ensures you're working with the most recent `contract-abi` that gets generated.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_abi"
+  commentType="//"
+  lang="rust"
+/>
+
+## Initializing Functions
+
+When writing tests for Sway, two crucial objects are required: the contract instance and the wallets that interact with it. This helper function ensures a fresh start for every new test case. It will export the deployed contracts, the contract ID, and all the generated wallets for this purpose.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_contract_instance_parent"
+  commentType="//"
+  lang="rust"
+/>
+
+### Contract Storage and Binary
+
+Besides the ABI, which outlines the interaction protocol with the smart contract, it's imperative to also load the contract's storage and binary. These three components are vital for the successful creation and deployment of the contract instance, guaranteeing precise testing in the following phases.
+
+<CodeImport
+  file="../../../intro-to-sway/contract/tests/harness.rs"
+  comment="rs_contract_instance_config"
+  commentType="//"
+  lang="rust"
+/>
+
+## Test Cases
+
+Given the immutable nature of smart contracts, it's not only vital to test basic functionalities but also imperative to thoroughly cover all potential edge cases.
```

Comment:
Clarify that in this section the user should be adding these functions to the bottom of harness.rs